app/services/storage.service.ts

```diff
@@ -199,7 +203,51 @@ export class StorageService {
      * @param options - Optional parameters
      */
     public deleteBlobIfExists(container: string, blob: string, options: any = {}): Observable<any> {
-        return this._callStorageClient((client) => client.deleteBlobIfExists(container, blob, options));
+        return this._callStorageClient((client) => {
+            return client.deleteBlobIfExists(container, blob, options).then((result) => {
+                const blobCache = this.getBlobFileCache({ container: container });
+                if (result && blobCache) {
+                    // cache key is file.name (the name of the blob excluding the container)
+                    blobCache.deleteItemByKey(blob);
+                }
+            });
+        });
+    }
+
+    /**
+     * Delete the list of files from the container in storage
+     * @param container - Instance of the container
+     * @param files - Files to delete
+     * @param options - Optional parameters
+     */
+    public deleteFilesFromContainer(container: BlobContainer, files: File[], options: any = {}) {
+        const fileCount = files.length;
+        const taskTitle = `Delete ${fileCount} files from ${container.name}`;
+        let deleted = 0;
+
+        return this.backgroundTaskService.startTask(taskTitle, (task) => {
+            // NOTE: slight pause in-between deletes to ease load on storage account
+            const observable = Observable.interval(100).take(fileCount);
+            observable.subscribe({
+                next: (i) => {
+                    deleted++;
```

Comment:
isn't deleted the same as i?

Reply:
yes, though i need to change it to be the actual number of deleted files.

Reply:
but actually what do you expect to show if the number is different? Some files wasn't deleted successfully or some files were already deleted?

Reply:
i expected i would show a warning that not all files were able to be deleted. but maybe i just shouldn't bother ....